firmware/application/apps/ui_recon.cpp

```diff
@@ -93,15 +122,15 @@ void ReconView::colorize_waits() {
 bool ReconView::recon_save_freq(const std::string& freq_file_path, size_t freq_index, bool warn_if_exists) {
     File recon_file;
 
-    if (frequency_list.size() == 0 || (frequency_list.size() && current_index > (int32_t)frequency_list.size()))
+    if (frequency_list.size() == 0 || !current_is_valid())
         return false;
 
-    freqman_entry entry = frequency_list[freq_index];
+    freqman_entry entry = *frequency_list[freq_index];  // Makes a copy.
     entry.frequency_a = freq;
     entry.frequency_b = 0;
     entry.modulation = last_entry.modulation;
     entry.bandwidth = last_entry.bandwidth;
-    entry.type = SINGLE;
+    entry.type = freqman_type::Single;
```

Comment:
I supposed we could add a using statement at the top if you don't want to see all the qualified enum names.

Reply:
I'm okay with this